packages/data/src/redux-store/metadata/reducer.ts

```diff
@@ -36,17 +40,41 @@ const subKeysIsResolved: Reducer< Record< string, State >, Action > = onSubKey<
 	switch ( action.type ) {
 		case 'START_RESOLUTION':
 		case 'FINISH_RESOLUTION': {
-			const isStarting = action.type === 'START_RESOLUTION';
+			const isResolving = action.type === 'START_RESOLUTION';
```

Comment:
I'd personally split both cases into separate chunks of code:
```js
case START:
  ...
case FINISH:
  ...
case FAIL:
  ...
```
The `nextState = new ...` and `return nextState` statements are repeated, but the whole thing is easier to read IMO.